src/gluonts/dataset/pandas.py

```diff
@@ -81,53 +100,59 @@ class PandasDataset:
     feat_static_cat: List[str] = field(default_factory=list)
     past_feat_dynamic_real: List[str] = field(default_factory=list)
     ignore_last_n_targets: int = 0
+    unchecked: bool = False
+    assume_sorted: bool = False
 
     def __post_init__(self) -> None:
         if isinstance(self.target, list) and len(self.target) == 1:
             self.target = self.target[0]
         self.one_dim_target = not isinstance(self.target, list)
 
-        if is_series(self.dataframes):
-            self.dataframes = series_to_dataframe(self.dataframes)
-        # store data internally as List[Tuple[str, pandas.DataFrame]]
-        # if str is not empty it will be set in ``DataEntry`` as ``item_id``.
+        if isinstance(self.dataframes, (pd.Series, pd.DataFrame)):
+            self.dataframes = [self.dataframes]
+
+        assert isinstance(self.dataframes, SizedIterable)
+
         if isinstance(self.dataframes, dict):
-            self._dataframes = list(self.dataframes.items())
-        elif isinstance(self.dataframes, list):
-            self._dataframes = [(None, df) for df in self.dataframes]
-        else:  # case single dataframe
-            self._dataframes = [(None, self.dataframes)]
+            self._pairs: Any = self.dataframes.items()
```

Comment:
Anything that is assigned to should be listed as a definition.

Reply:
not sure I get it